x-pack/plugins/observability/server/lib/rules/slo_burn_rate/executor.ts

```diff
@@ -172,40 +172,40 @@ export const getRuleExecutor = ({
           scheduledActionsCount++;
         }
       }
+    }
+    alertFactory.alertLimit.setLimitReached(hasReachedLimit);
 
-      const { getRecoveredAlerts } = alertFactory.done();
```

Comment:
This actually fixed the bug. I had to move this part outside of the alerts loop. 

We were checking for recovered alerts inside the loop of alert results. So, when there were no alerts anymore, this part of code that is setting the context to the action connector would never run. That's why the slo context on Slack was empty.

Reply:
👏🏻 sneaky bug, nice found!